tests/openssh.rs

```diff
@@ -35,6 +36,37 @@ async fn control_dir() {
     std::fs::remove_dir(&dirname).unwrap();
 }
 
+#[tokio::test]
+#[cfg_attr(not(ci), ignore)]
+async fn config_file() {
```

Comment:
For this test, let's check that the connection succeeds to `addr()` rather than it failing to a random localhost port. You _should_ be able to just write the value returned from `addr()` into the generated config for `config-file-test`, and then connect normally just like in the `it_connects` tests.

Reply:
Updated in next commit.

Reply:
looks like this test now passes with the latest update to port 2222: 

2021-07-13T00:09:01.3082187Z test config_file ... ok

Tarpaulin is still failing, but I am not sure that's caused by this change - was it failing previous to this pull request?

Reply:
So, there is a known problem with tarpaulin on CI here, but in this case the problem is specifically that you're hard-coding `127.0.0.1`, which won't always be right on CI. You should instead use the value provided by calling `addr()` (see the other tests).

Reply:
Are you proposing that I parse the addr() string like "ssh://my-user@127.0.0.1:8888" into it component parts and then write the ssh_config file like this generated from it?

   let ssh_config_contents = r#"Host config-file-test
        User my-user
        HostName 127.0.0.1
        Port 8888"#;

Or were you thinking of a one-liner replacement in the ssh_config file with a simpler to parse syntax? I'm fine with writing either, I just don't want to add parsing to the tests if that doesn't need to be added.

Reply:
I took a whack at what you think you had in mind in the next revision below...regexes are hard, so I ended up writing tests for my tests...:) Open to a better option than that, but I suspect it'll get the job done.